inc/datahandlers/post.php

```diff
@@ -252,28 +252,60 @@ function verify_message()
 			$this->set_error("missing_message");
 			return false;
 		}
-
-		// If this board has a maximum message length check if we're over it. Use strlen because SQL limits are in bytes
-		else if(strlen($post['message']) > $mybb->settings['maxmessagelength'] && $mybb->settings['maxmessagelength'] > 0 && !is_moderator($post['fid'], "", $post['uid']))
+		else
 		{
-			$this->set_error("message_too_long", array($mybb->settings['maxmessagelength'], strlen($post['message'])));
-			return false;
-		}
+			$limit = (int)$mybb->settings['maxmessagelength'];
```

Comment:
Here you should add the is_moderator condition:
```php
			if(!is_moderator($post['fid'], '', $post['uid']))
			{
				$limit = (int)$mybb->settings['maxmessagelength'];
			}
			else
			{
				$limit = 0;
			}
```

Reply:
Logic updated. Thanks for review.